assets/js/util/sort-object-map-by-key.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * sortObjectMapByKey utiltity function.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Sorts an object map based on numberic order/priority key.
+ *
+ * @param {Object} objectList  Object containg objects to be sorted
+ * @param {string} priorityKey The object key to use for matching.
+ *
+ * @return {Array} An Array containing objects ordered by the priority key
+ */
+export function sortObjectMapByKey( objectList, priorityKey ) {
```

Comment:
It's a bit weird to have this accept an object and return an array, especially since in some instances above we're already passing an array to it I believe? Does this not fail as is?

Either way, I think we should either make this accept an array and return an array (transforming back and forth from object to array would be handled outside the function) or accept an object and return an object (transforming is handled completely by the function). Since we already use `Object.values` on these objects for other reasons too (e.g. filtering), I think it makes sense to simplify this function to just sort arrays. It also means the function fulfils only a single purpose which I think is cleaner.